plugins/robothor_plugin/robothor_tasks.py

```diff
@@ -253,7 +253,7 @@ def _step(self, action: Union[int, Sequence[int]]) -> RLStepResult:
 
         if action_str == END:
             self._took_end_action = True
-            self._success = self._is_goal_in_range()
+            self._success = self._is_goal_in_range() and self._is_goal_visible()
```

Comment:
Yah, I understand the motivation for doing this but I think we should keep using THOR's definition of visibility rather than defining our own.